lib/chat_api/reporting.ex

```diff
@@ -40,6 +40,26 @@ defmodule ChatApi.Reporting do
     |> Repo.all()
   end
 
+  def average_first_replied_time(account_id, filters \\ %{}) do
+    Conversation
+    |> where(account_id: ^account_id)
+    |> where(^filter_where(filters))
+    |> where([conv], not is_nil(conv.first_replied_at))
+    |> select([:first_replied_at, :inserted_at])
+    |> Repo.all()
+    |> compute_average_replied_time()
+  end
+
+  def compute_average_replied_time(conversations) do
+    conversations
+    |> Enum.map(fn conv -> Time.diff(conv.first_replied_at, conv.inserted_at) end)
+    |> average()
+  end
+
+  def average(list) do
+    Enum.sum(list) / max(length(list), 1)
```

Comment:
not important but for the sake of learning elixir -- one common technique people use in elixir is pattern matching to handle special cases (e.g. an empty list, which in this case would result in a denominator of zero)

for example, in this case we could write this like
```elixir
@spec average([integer()]) :: float()
def average([]), do: 0

def average(list) do
  Enum.sum(list) / length(list)
end
```

(just a useful pattern to be aware of, since elixir doesn't support early returns)

Reply:
Sounds good thanks!